docs/proposals/diagnose-experience.md

```diff
@@ -0,0 +1,250 @@
+> Request method: "GET"
+> Request headers:
+   "User-Agent": "oras/1.2.0+Homebrew"
+   "Accept": "application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.oci.image.manifest.v1+json, application/vnd.oci.image.index.v1+json, application/vnd.oci.artifact.manifest.v1+json" 
+
+
+[2024-08-02 23:56:03] <-- Response #0
+< Response Status: "401 Unauthorized"
+< Response headers:
+   "Content-Length": "73"
+   "X-Github-Request-Id": "9FC6:30019C:17C06:1C462:66AD0463"
+   "Content-Type": "application/json"
+   "Www-Authenticate": "Bearer realm=\"https://ghcr.io/token\",service=\"ghcr.io\",scope=\"repository:oras-project/oras:pull\""
+   "Date": "Fri, 02 Aug 2024 23:56:03 GMT"
+< Response body:
+{
+        "errors": [
+            {
+                "code": "<UNAUTHORIZED>",
+                "message": "<message describing condition>",
+                "detail": "<unstructured>"
+            },
+            ...
+        ]
+}
+
+
+[2024-08-02 23:56:04] --> Request #1
+> Request URL: "https://ghcr.io/token?scope=repository%3Aoras-project%2Foras%3Apull&service=ghcr.io"
+> Request method: "GET"
+> Request headers:
+   "User-Agent": "oras/1.2.0+Homebrew" 
+
+
+[2024-08-02 23:56:04] <-- Response #1
+< Response Status: "200 OK"
+< Response headers:
+   "Content-Type": "application/json"
+   "Docker-Distribution-Api-Version": "registry/2.0"
+   "Date": "Fri, 02 Aug 2024 16:08:05 GMT"
+   "Content-Length": "69"
+   "X-Github-Request-Id": "9FC6:30019C:17C0D:1C46C:66AD0464" 
+<  Response body:
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:42c524c48e0672568dbd2842d3a0cb34a415347145ee9fe1c8abaf65e7455b46",
+      "size": 1239,
+      "platform": {
+        "architecture": "amd64",
+        "os": "linux"
+      }
+    },
+    ···
+}
+
+
+```
+
+### oras push/pull/attach
+
+The verbose output of commands `pull`, `push`, `attach` is now printed out in the default (status) output, the `--verbose` is no longer needed. Considering the progress bar is shown on terminal by default, the verbose output should be available on non-terminal environment using `--no-tty`. See `oras pull` command as an example:
+
+```bash
+$ oras push --oci-layout layout-test:sample README.md --no-tty
+
+Preparing README.md
+Uploading 9500d720111f README.md
+Uploading 44136fa355b3 application/vnd.oci.empty.v1+json
+Uploaded  44136fa355b3 application/vnd.oci.empty.v1+json
+Uploaded  9500d720111f README.md
+Uploading 655f5cc5d5d2 application/vnd.oci.image.manifest.v1+json
+Uploaded  655f5cc5d5d2 application/vnd.oci.image.manifest.v1+json
+Pushed [oci-layout] layout-test:sample
+ArtifactType: application/vnd.unknown.artifact.v1
+Digest: sha256:655f5cc5d5d2e7ff2ab90378514103996523b065ce5ef43cd6e6a4de7d80a535
+```
+
+### Show user's environment details
```

Comment:
Looks like this is not included in the Proposal section?

Reply:
Updated